src/reducers/ByProgramReducer.js

```diff
@@ -125,6 +125,7 @@ export const byProgramReducer = (state, action) => {
       return {
         ...state,
         supplier: value,
+        program: null,
```

Comment:
Because of the change of ordering, once selecting a supplier, set program to null